### PR TITLE
CORE-16320 fix state-manager config

### PR DESCRIPTION
**Pull request by conal-smith**
And reinstate FlowMaintenance component

### CODE REVIEW

File: libs/state-manager/state-manager-db-impl/src/main/kotlin/net/corda/libs/statemanager/impl/repository/impl/PostgresQueryProvider.kt [hunk 1]
```diff
@@ -5,37 +5,41 @@ import net.corda.db.schema.DbSchema.STATE_MANAGER_TABLE
...
 
 class PostgresQueryProvider : AbstractQueryProvider() {
+    private companion object {
+        private const val STATE_MANAGER_SCHEMA = "state_manager"
+    }
+
     override val createState: String
         get() = """
-            INSERT INTO $STATE_MANAGER_TABLE
+            INSERT INTO $STATE_MANAGER_SCHEMA.$STATE_MANAGER_TABLE
```
Comment (sprintengine) [inline libs/state-manager/state-manager-db-impl/src/main/kotlin/net/corda/libs/statemanager/impl/repository/impl/PostgresQueryProvider.kt]: Seemingly part of the problem, not specifying the schema in these native SQL queries

File: libs/state-manager/state-manager-db-impl/src/main/kotlin/net/corda/libs/statemanager/impl/repository/impl/PostgresQueryProvider.kt [hunk 2]
```diff
@@ -5,37 +5,41 @@ import net.corda.db.schema.DbSchema.STATE_MANAGER_TABLE
 import net.corda.libs.statemanager.api.MetadataFilter
 
 class PostgresQueryProvider : AbstractQueryProvider() {
+    private companion object {
+        private const val STATE_MANAGER_SCHEMA = "state_manager"
```
Comment (sprintengine) [inline libs/state-manager/state-manager-db-impl/src/main/kotlin/net/corda/libs/statemanager/impl/repository/impl/PostgresQueryProvider.kt]: We should bring this in via the constructor injection, and it should come from the config

File: libs/state-manager/state-manager-db-impl/src/integrationTest/kotlin/net/corda/libs/statemanager/impl/tests/StateManagerIntegrationTest.kt [hunk 3]
```diff
@@ -42,12 +42,18 @@ import java.util.UUID
...
 // TODO-[CORE-16663]: make database provider pluggable
+@Disabled
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class StateManagerIntegrationTest {
 
-    private val dbConfig: EntityManagerConfiguration = DbUtils.getEntityManagerConfiguration("state_manager_db")
+    private val dbConfig: EntityManagerConfiguration = DbUtils.getEntityManagerConfiguration(
+        inMemoryDbName = "state_manager_db",
+        schemaName = "state_manager",
+        createSchema = true
```
Comment (sprintengine) [inline libs/state-manager/state-manager-db-impl/src/integrationTest/kotlin/net/corda/libs/statemanager/impl/tests/StateManagerIntegrationTest.kt]: I'll re-run with this test enabled and check if this "createSchema" works.

EDIT - this appears to work locally

**Review by jujoramos — APPROVED** [review]
LGTM

**Pull request merged**